apax/cli/templates/train_config_full.yaml

```diff
@@ -16,8 +16,11 @@ data:
   #train_data_path: <PATH>
   #val_data_path: <PATH>
   #test_data_path: <PATH>
+  dataset:
+    name: cached
```

Comment:
I think I'd prefer `type` over `name`. But I see why you don't want to use that as attribute. Maybe `method` or `processing`?

Reply:
I used name to be consistent with the other uses of discriminative unions in our config. 
I think method would be fine but I think we should only use one keyword for this purpose

Reply:
I'll change it to processing

Reply:
My reasoning against `name` was, that `dataset:name` reminds me of `train`, `test` or `SPICE` etc.